### PR TITLE
Refactors the russian revolver, and you can no longer play RR without a head

### PR DESCRIPTION
**Pull request by Citinited**
This PR does the following:
- Refactors the russian revolver so it's slightly more idiot-proof;
- Makes it so headless people can't play RR.
- You can now hit other humans with the RR (force of 5, so it's not a very powerful weapon)

:cl:
tweak: You can't play russian roulette without a head
tweak: using a russian revolver on another human will now hit them with the revolver instead of failing to play russian roulette with them.
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
I guess this might fix #8699?

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1381,35 +1380,38 @@ obj/item/toy/cards/deck/syndicate/black
...
+	if(!(user.has_organ("head"))) //For sanity
+		to_chat(user, "<span class='notice'>Playing this game without a head would be classed as cheating.</span>")
+		return
+	user.visible_message("<span class='danger'>[user] points [src] at their head, ready to pull the trigger!</span>")
+	if(do_after(user, 30, target = user))
+		if(bullet_position > 1)
 			user.visible_message("<span class='danger'>*click*</span>")
 			playsound(src, 'sound/weapons/empty.ogg', 100, 1)
-			bullet_position -= 1
+			bullet_position --
```
Comment (tigercat2000) [inline code/game/objects/items/toys.dm]: No spacing between vars and -- `bullet_position--`

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1381,35 +1380,38 @@ obj/item/toy/cards/deck/syndicate/black
...
-		if(bullet_position == 1)
-			is_empty = 1
+		else
+			bullet_position = null
 			playsound(src, 'sound/weapons/Gunshot.ogg', 50, 1)
-			user.visible_message("<span class='danger'>The [src] goes off!</span>")
-			M.apply_damage(200, BRUTE, "head", sharp =1, used_weapon = "Self-inflicted gunshot wound to the head.")
-			M.death()
+			user.visible_message("<span class='danger'>[src] goes off!</span>")
+			user.apply_damage(200, BRUTE, "head", sharp =1, used_weapon = "Self-inflicted gunshot wound to the head.")
```
Comment (tigercat2000) [inline code/game/objects/items/toys.dm]: Spacing `sharp = 1`

**Comment by Citinited** [comment]
@tigercat2000 Changes committed

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1381,35 +1380,38 @@ obj/item/toy/cards/deck/syndicate/black
...
 		spin_cylinder()
-		user.visible_message("<span class='warning'>[user] spins the cylinder on the [src]!</span>")
+		user.visible_message("<span class='warning'>[user] spins the cylinder on [src]!</span>")
 
 
-/obj/item/toy/russian_revolver/attack(mob/living/carbon/M, mob/user)
-	if(M != user) //can't use this on other people
+/obj/item/toy/russian_revolver/attack(atom/target, mob/living/carbon/human/user)
+	if(target != user) //can't use this on other people
 		return
```
Comment (Citinited) [inline code/game/objects/items/toys.dm]: maybe this should `return ..()` 🤔

**Comment by Citinited** [comment]
Oh wait, it needs an attack sound as it is currently silent. I'll add it after I return from work today.

**Pull request merged**